WooCommerce/Classes/ViewRelated/Inbox/InboxNoteRowViewModel.swift

```diff
@@ -34,21 +49,72 @@ struct InboxNoteRowViewModel: Identifiable, Equatable {
             return formatter.localizedString(for: note.dateCreated, relativeTo: today)
         }()
         let actions = note.actions.map { InboxNoteRowActionViewModel(action: $0) }
+
         self.init(id: note.id,
                   date: date,
                   typeIcon: (NoteType(rawValue: note.type) ?? .info).image,
                   title: note.title,
                   attributedContent: attributedContent,
-                  actions: actions)
+                  actions: actions,
+                  siteID: note.siteID,
+                  stores: stores,
+                  storageManager: storageManager)
     }
 
-    init(id: Int64, date: String, typeIcon: Image, title: String, attributedContent: NSAttributedString, actions: [InboxNoteRowActionViewModel]) {
+    init(id: Int64,
+         date: String,
+         typeIcon: Image,
+         title: String,
+         attributedContent: NSAttributedString, actions: [InboxNoteRowActionViewModel],
+         siteID: Int64,
+         stores: StoresManager = ServiceLocator.stores,
+         storageManager: StorageManagerType = ServiceLocator.storageManager) {
         self.id = id
         self.date = date
         self.typeIcon = typeIcon
         self.title = title
         self.attributedContent = attributedContent
         self.actions = actions
+        self.siteID = siteID
+        self.stores = stores
+        self.storageManager = storageManager
+    }
+
+    static func == (lhs: InboxNoteRowViewModel, rhs: InboxNoteRowViewModel) -> Bool {
+        lhs.id == rhs.id &&
+        lhs.siteID == rhs.siteID
+    }
+}
+
+extension InboxNoteRowViewModel {
+    func markInboxNoteAsActioned() {
+        guard let actionID = actions.first?.id else {
+            return
+        }
```

Comment:
I've seen notes with more than one action during testing, it'd be safer passing the action ID here:
```suggestion
    func markInboxNoteAsActioned(actionID: Int64) {
```